src/components/resume-builder/context/ResumeDataContext.tsx

```diff
@@ -0,0 +1,207 @@
+  ],
+  workExperience: {
+    title: 'WORK EXPERIENCE',
+    experience: [
+      {
+        companyName: 'Google Inc.',
+        jobTitle: 'Software Engineer',
+        startDate: 'June 2023',
+        endDate: 'Present',
+        description: ['Worked on YouTube Ads Scaling.'],
+      },
+    ],
+  },
+  projects: {
+    title: 'PROJECTS',
+    projects: [
+      {
+        organizationName: 'Google Cloud.',
+        projectTitle: 'To do App',
+        startDate: 'June 2022',
+        endDate: 'Aug 2022',
+        description: ['Worked on YouTube Ads Scaling.'],
+      },
+    ],
+  },
+  education: {
+    title: 'EDUCATION',
+    courses: [
+      {
+        courseName: 'Bachelor of Science in Computer Science',
+        institutionName: 'University of Technology',
+        startDate: 'May 2019',
+        endDate: 'May 2023',
+        scoreEarned: '9.23 GPA',
+        description:
+          'Specialized in Artificial Intelligence and Data Science. Completed a capstone project on machine learning algorithms for predictive analytics.',
+      },
+      {
+        courseName: 'Intermediate PCM',
+        institutionName: 'Central High School',
+        startDate: 'Mar 2016',
+        endDate: 'Mar 2018',
+        scoreEarned: '93.4%',
+        description:
+          'Valedictorian. President of the Computer Science Club. Participated in national mathematics olympiad.',
+      },
+    ],
+  },
+  activities: {
+    title: 'ACTIVITIES',
+    activities: [
+      {
+        activityTitle: 'Student Council President',
+        institutionName: 'University of Example',
+        startDate: 'Sept 2021',
+        endDate: 'May 2022',
+        descriptions: [
+          'Led weekly meetings to address student concerns',
+          'Organized campus-wide events with over 1000 participants',
+        ],
+      },
+    ],
+  },
+  skills: {
+    title: 'SKILLS',
+    skillSet: [
+      {
+        title: 'Programming Languages',
+        skills: ['JavaScript', 'TypeScript', 'Python', 'Java', 'C++', 'SQL'],
+      },
+      {
+        title: 'Soft Skills',
+        skills: [
+          'Team Leadership',
+          'Agile Methodologies',
+          'Technical Writing',
+          'Public Speaking',
+        ],
+      },
+    ],
+  },
+  achievements: {
+    title: 'ACHIEVEMENTS',
+    achievementList: [
+      {
+        awardName: 'Best Paper Award',
+        institutionName: 'International Conference on Computer Science',
+        dateAwarded: 'May 2023',
+        description:
+          "Received for the paper 'Novel Approaches in Machine Learning'",
+      },
+    ],
+  },
+};
+
+export const ResumeDataProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  /**
+   * States to save the Resume Data.
+   * At the same time, I decided to keep the states seperate and not a consolidated object, because updating the state can be a tedious task for sections like Work Experience, etc that have big complex objects.
+   */
```

Comment:
Also write, state objects for other sections are not needed when we are updating let's say 'title' section.